File: scala-code/src/main/scala/com/springernature/slides/Slide3.scala
package com.springernature.slides

object Slide3 {
  class Specie(
    val kind: String
  ) {
    def eats(): Seq[String] = Seq()
  }

  class Person(
    val name: String,
    var age: Int = 0
  ) extends Specie("Human") {
    override def eats() = Seq("...")
  }
}

File: kotlin-code/src/main/kotlin/com/springernature/slides/Slide3.kt
package com.springernature.slides

open class Specie(
    val kind: String
) {
    open fun eats(): List<String> = emptyList()
}

class Person3(
    val name: String,
    var age: Int = 0
): Specie("Human") {
    override fun eats() = listOf("...")
}